config/lib/logger.js

```diff
@@ -0,0 +1,46 @@
+﻿'use strict';
+
+var _ = require('lodash'),
+  config = require('../config'),
+  fs = require('fs');
+
+// build logger service
+var logger = {
+  getFormat: getLogFormat, // log format to use
+  getOptions: getLogOptions // log options to use
+};
+
+// export the logger service
+module.exports = logger;
+
+/**
+ * The format to use with the logger
+ *
+ * Returns the log.format option set in the current environment configuration
+ */
+function getLogFormat () {
+  return config.log && config.log.format ? config.log.format : 'combined';
+}
+
+/**
+ * The options to use with the logger
+ *
+ * Returns the log.options object set in the current environment configuration.
+ * NOTE: Any options, requiring special handling (e.g. 'stream'), that encounter an error will be removed from the options.
+ */
+function getLogOptions () {
+  var options = config.log && config.log.options ? config.log.options : {};
+
+  // check if the current environment config has the log stream option set
+  if (_.has(options, 'stream')) {
+    try {
+      // create the WriteStream to use for the logs
+      options.stream = fs.createWriteStream(process.cwd() + '/' + config.log.options.stream, { flags: 'a' });
+    } catch (err) {
+      // remove the stream option
+      delete options.stream;
```

Comment:
Do we want to log an error to the console instead of just silently catching and handeling the error?